book/src/SDK/platform/mongodb/overview.md

```diff
@@ -0,0 +1,44 @@
+## MongoDB
+
+The MongoDB module is an opinionated micro-framework for interacting with the MongoDB backend that 
+Harbor currently supports. This module was developed based on the following needs and desires:
+
+- At the time of initial development there did not seem to be a widely-used or well-maintained 
+  Rust-based Object Document Mapper (ODM) available from the ecosystem. Database mapping code is 
+  inherently full of boilerplate, which we equate to engineering toil, and can also be a 
+  significant source of bugs. While the Rust type system helps with the latter issue to a great 
+  degree, we still wanted to leverage generics to generate as much boilerplate code as possible.
+- The official Mongo Rust Driver (see below) is wonderful, but also deep and finely-grained. 
```

Comment:
OK, I don't know what this means:
> The official Mongo Rust Driver (see below) is wonderful, but also deep and finely-grained

Saying that it's wonderful is rather subjective, no?  Also, what is a "deep and finely-grained" actually mean?  That the driver is what I would call "Low Level"?  Where there aren't many niceties?

Reply:
It is wonderful! I like to praise people when possible. 😁 

Yeah, I think "deep and finely-grained" can be summarized as low-level.  The point is it exposes most if not all the possible options, and requires you to specify them.  

I've updated this to just the following 

"The official Mongo Rust Driver (see below) is a low-level driver designed to allow total control of 
  all aspects of MongoDB interaction. We wanted a way to abstract away that complexity and allow 
  developers to focus on domain logic rather than repeatedly handle infrastructure concerns."